### PR TITLE
[Benchmark] New benchmarking tool with gnuplot(2)

### PR DESCRIPTION
**Pull request by shibatch**
This is a rebased branch of https://github.com/shibatch/sleef/pull/98.

This patch replaces the old benchmarking tool with a new tool with gnuplot.

With this new tool, the graph showing execution time can be automatically drawn.
It is now easy to see the difference between different library versions or compilers.
This tool is used to draw the graphs on sleef.org web site.

This patch implements the tool described in https://github.com/shibatch/sleef/issues/9.

A makefile is used In this patch, and there are a few reasons.
* It needs to compile the tools for SVML. These tools are independent of the other part, and thus it does not make sense to build these tools using the cmake during a build of library.
* This makefile is used as a script for measurement. It is handy since ProcData.java has to be also compiled.

### CODE REVIEW

File: src/Makefile [hunk 1]
```diff
@@ -5,6 +5,7 @@ clean :
 	+"$(MAKE)" --directory=./gencoef clean
 	+"$(MAKE)" --directory=./libm clean
 	+"$(MAKE)" --directory=./libm-tester clean
+	+"$(MAKE)" --directory=./libm-benchmarks clean
```
Comment (fpetrogalli-arm) [inline src/Makefile]: Remove, we don't care about makefiles.

File: src/libm-benchmarks/Makefile [hunk 2]
```diff
@@ -0,0 +1,6 @@
+.PHONY: clean
```
Comment (fpetrogalli-arm) [inline src/libm-benchmarks/Makefile]: Remove, we don't care about makefiles.

File: src/libm-benchmarks/plot-scripts/Makefile [hunk 3]
```diff
@@ -0,0 +1,110 @@
+BENCHSLEEF := $(shell command -v benchsleef 2> /dev/null)
```
Comment (fpetrogalli-arm) [inline src/libm-benchmarks/plot-scripts/Makefile]: Remove, no makefiles.

File: src/libm-benchmarks/plot-scripts/benchsvml.c [hunk 4]
```diff
@@ -0,0 +1,153 @@
+//          Copyright Naoki Shibata 2010 - 2017.
```
Comment (fpetrogalli-arm) [inline src/libm-benchmarks/plot-scripts/benchsvml.c]: This looks like a benchmark, why is it in the plot folder? The plot-scripts folder should just contain the scripts that are used to make the plots.

File: src/libm-benchmarks/plot-scripts/benchsvml128.c [hunk 5]
```diff
@@ -0,0 +1,137 @@
+//          Copyright Naoki Shibata 2010 - 2017.
```
Comment (fpetrogalli-arm) [inline src/libm-benchmarks/plot-scripts/benchsvml128.c]: Another benchmark that should sit one folder up?

File: src/libm-benchmarks/plot-scripts/benchsvml256.c [hunk 6]
```diff
@@ -0,0 +1,140 @@
+//          Copyright Naoki Shibata 2010 - 2017.
```
Comment (fpetrogalli-arm) [inline src/libm-benchmarks/plot-scripts/benchsvml256.c]: Same, benchmark code, not script for plotting.

File: src/libm-benchmarks/plot-scripts/benchsvml512.c [hunk 7]
```diff
@@ -0,0 +1,137 @@
+//          Copyright Naoki Shibata 2010 - 2017.
```
Comment (fpetrogalli-arm) [inline src/libm-benchmarks/plot-scripts/benchsvml512.c]: Same, benchmark, not script.

File: src/libm-benchmarks/plot-scripts/measure.sh [hunk 8]
```diff
@@ -0,0 +1,17 @@
+#!/bin/sh
+echo
+read -p "Enter label of measurement(e.g. My desktop PC) : " label
+
+if [ -f counter.txt ]
+then
+    counter=`cat counter.txt`
```
Comment (fpetrogalli-arm) [inline src/libm-benchmarks/plot-scripts/measure.sh]: Does counter need any particular syntax?

File: src/libm-benchmarks/plot-scripts/measure.sh [hunk 9]
```diff
@@ -0,0 +1,17 @@
+#!/bin/sh
```
Comment (fpetrogalli-arm) [inline src/libm-benchmarks/plot-scripts/measure.sh]: It is not clear how to use this script. Who is invoking the java tool?

File: src/libm-benchmarks/plot-scripts/Makefile (hunk 3 above)
Comment (shibatch) [reply src/libm-benchmarks/plot-scripts/Makefile]: Did you even read the comment at the top of this page?

Comment (fpetrogalli-arm) [reply src/libm-benchmarks/plot-scripts/Makefile]: Yes, I still think we shouldn't have makefiles down inside the tree of `src`.

If you really want to keep the makefiles, then I think you should move all this benchmarks and relative scripts at the same level of `src`, in a folder like `svml-benchmarking`, and keep it as a sort of independent project inside SLEEF.

Comment (shibatch) [reply src/libm-benchmarks/plot-scripts/Makefile]: Originally, the whole directory including all the C codes were like an independent project.
I think that is easier to understand.
Why do you think C codes have to be compiled by cmake?

**Comment by fpetrogalli-arm** [comment]
> Originally, the whole directory including all the C codes were like an independent project.
> I think that is easier to understand.
Agree. Please move it at the same level of `src`.

> Why do you think C codes have to be compiled by cmake?
I don't think it anymore. Please remove any cmake configuration related to these benchmark.

**Pull request merged**